catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/ModalRegistry.view.js

```diff
@@ -161,31 +145,44 @@ function (ich,Marionette,Backbone,ConfigurationEdit,wreqr,_,$,Utils,Service,moda
                 }
             });
         },
-        renderNameField: function() {
-            var model = this.model;
-            var $registryName = this.$(".registryName");
-            var initialName = model.get('name');
-            var data = {
-                id: model.id,
-                name: 'Registry Name',
-                defaultValue: [initialName],
-                description: 'Unique identifier for a remote registry configuration.'
-            };
-            $registryName.append(ich.textType(data));
-            $registryName.val(data.defaultValue);
-            Utils.setupPopOvers($registryName, data.id, data.name, data.description);
-        },
         registryNameChanged: function(evt) {
             var newName = this.$(evt.currentTarget).find('input').val().trim();
             this.checkName(newName);
         },
+        urlChanged: function(evt) {
+            var url = evt.target.value;
+            var regExUrl;
+            if (url) {
+                regExUrl = this.extractHostnameFromUrl(url);
+            }
+            var hostname;
+            if (regExUrl) {
+                hostname = regExUrl;
+                if (hostname) {
+                    this.checkName(hostname);
+                } else {
+                    this.checkName(url);
+                }
+            }
+            var $registryName = this.$(".registryName");
+            var $group = this.$el.find('.registryName>.control-group');
+            $group.empty();
+            $registryName.val(hostname);
+        },
+        extractHostnameFromUrl: function(url){
+            var myRe = new RegExp("^(?:([^:/?#.]+):)?(?://)?(([^:/?#]*)(?::(\\d*))?)((/(?:[^?#](?![^?#/]*\\.[^?#/.]+(?:[\\?#]|$)))*/?)?([^?#/]*))?(?:\\?([^#]*))?(?:#(.*))?");
+            var regExUrl = myRe.exec(url);
+            return regExUrl[2];
+        },
+        cancel: function(){
+            this.closeAndUnbind();
+        },
         checkName: function(newName) {
             var view = this;
             var model = view.model;
             var config = this.getConfig();
-
             if (newName === '') {
-                view.showError('A registry must have a unique name.');
+                view.showError('A registry must have a unique URL.');
```

Comment:
This error message is fine, but I would consider re-wording it to say, something like, "A registry must have a URL". since the uniqueness check is already taken being done below.